exporter/containerimage/writer.go

```diff
@@ -57,6 +57,13 @@ func (ic *ImageWriter) Commit(ctx context.Context, inp exporter.Source, sessionI
 		return nil, errors.Errorf("unable to export multiple refs, missing platforms mapping")
 	}
 
+	var p exptypes.Platforms
+	if ok {
+		if err := json.Unmarshal(platformsBytes, &p); err != nil {
```

Comment:
This doesn't seem to be backward compatible. The case is when new daemon(exporter) is used with an old frontend that doesn't set this key(because it is not exporting a manifest list).

Reply:
Would a length check here satisfy this concern? It's already not doing this unmarshal if the metadata is missing entirely.

Reply:
Updated to include that, also tested against some already released docker/dockerfile versions as the frontend, seems to be ok.